src/lib/input/input.ts

```diff
@@ -90,14 +95,46 @@ export class MdHint {
   @Input() align: 'start' | 'end' = 'start';
 }
 
+/**
+ * Enables auto expanding behaviour for the textarea
+ */
+@Directive({
+  selector: 'textarea[mdAutosize]'
+})
+export class MdAutosize implements OnInit {
+  @HostListener('input', ['$event.target'])
+  public onChange() {
+    if (this._isActive) {
+      this._adjust();
+    }
+  }
+
+  @Input('mdAutosize')
+  private _isActive: boolean;
+
+  constructor(private _elRef: ElementRef) {}
+
+  ngOnInit() {
+    if (this._isActive) {
+      this._adjust();
+    }
+  }
+
+  private _adjust() {
+    this._elRef.nativeElement.style.overflow = 'hidden';
+    this._elRef.nativeElement.style.height = 'auto';
+    this._elRef.nativeElement.style.height = `${this._elRef.nativeElement.scrollHeight}px`;
```

Comment:
The logic for sizing the textarea is much more complicated in material1, which makes me think this may be missing something.

@crisbeto do you have insights on this?